src/main/webapp/app/exercises/programming/manage/update/update-components/grading/programming-exercise-grading.component.ts

```diff
@@ -60,13 +64,16 @@ export class ProgrammingExerciseGradingComponent implements AfterViewInit, OnDes
     }
 
     calculateFormStatus() {
-        this.formValid = Boolean(
+        const newFormValidValue = Boolean(
             this.maxScoreField?.valid &&
                 this.bonusPointsField?.valid &&
                 (this.maxPenaltyField?.valid || !this.programmingExercise.staticCodeAnalysisEnabled) &&
                 !this.submissionPolicyUpdateComponent?.invalid &&
                 this.lifecycleComponent?.formValid,
         );
+
+        this.formValidSignal.set(newFormValidValue);
+        this.formValid = newFormValidValue;
```

Comment:
Yes, this would require quite some changes and bloat up the changes of this PR even more, which has already reached quite a size.
It is also not the point of this PR, I introduced signals wherever possible with reasonable effort at the components that I touched.

The new signal here is needed to make sure the StatusBar is reactive.